Chatgpt-plugin/azure-function/ext/SPASS/man/man1/dfg2otter.1
.\" Automatically generated by Pod::Man 2.1801 (Pod::Simple 3.05)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.ie \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.el \{\
.    de IX
..
.\}
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "DFG2OTTER 1"
.TH DFG2OTTER 1 "2010-03-18" "perl v5.10.0" "SPASS"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
dfg2otter \- transforms DFG clause files into Otter format
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
\&\fBdfg2otter\fR [options] <infile> <outfile>
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
\&\fBdfg2otter\fR is a C\-program to transform problem
input files in \fB\s-1DFG\s0\fR syntax into \fBOtter\fR syntax. It accepts
all options from \fB\s-1SPASS\s0\fR, although only a subset has an effect
on translation.
.PP
\&\fBdfg2otter\fR negates conjecture formulae of the \fB\s-1SPASS\s0\fR input file
before printing the Otter usable list. The \fB\s-1SPASS\s0\fR conjecture formula list
is translated into a disjunction of the negated single conjectures.
If the \fB\s-1SPASS\s0\fR input file consits
of clauses, these are not modified.
.SH "SEE ALSO"
.IX Header "SEE ALSO"
\&\fIcheckstat\fR\|(1), \fIfilestat\fR\|(1), \fIpcs\fR\|(1), \fIpgen\fR\|(1), \fIrescmp\fR\|(1), \fItpform\fR\|(1), \fItpget\fR\|(1), \fIdeprose\fR\|(1), \fIdfg2otter.pl\fR\|(1), \s-1\fISPASS\s0\fR\|(1)
.SH "AUTHORS"
.IX Header "AUTHORS"
Thomas Hillenbrand, Dalibor Topic and Christoph Weidenbach
.PP
Contact : spass@mpi\-inf.mpg.de


Chatgpt-plugin/azure-function/ext/SPASS/dfg2otter.pl
#! /usr/bin/perl

#**************************************************************
#* ********************************************************** *
#* *                                                        * *
#* *          Adds appropriate options for Otter            * *
#* *                                                        * *
#* *                                                        * *
#* $Revision: 1.1 $                                        * *
#* $State: Exp $                                            * *
#* $Date: 2006/03/07 16:19:26 $                             * *
#* $Author: topic $                                       * *
#* *                                                        * *
#* *             Contact:                                   * *
#* *             Christoph Weidenbach                       * *
#* *             MPI fuer Informatik                        * *
#* *             Stuhlsatzenhausweg 85                      * *
#* *             66123 Saarbruecken                         * *
#* *             Email: spass@mpi-inf.mpg.de                * *
#* *             Germany                                    * *
#* *                                                        * *
#* ********************************************************** *
#**************************************************************


# $RCSfile: dfg2otter.pl,v $ 


#############################################################
#
# Perl script for converting DFG input files into
# OTTER input files. The main work is done by the
# c-executable 'dfg2otter'.
#
# Usage:
#
# dfg2otter.pl [ -t] [file1] [file2]
#
# Options:
#
# -t (time): sets timelimit in seconds for OTTER in the converted
#     file (i.e., generates the command 'assign(max_seconds, time).')
#
# Functionality:
#
# The number of file arguments determines where dfg2otter
# takes its input from and directs its output to:
#
# file1 and file2 are given:
#     read from file1, write to file2
#
# file1 is given:
#     read from file1, generate output file file.ot
#
# No file arguments: 
#     input is read from stdin, output written to stdout
#
#
#############################################################

#########################################################
#  Packages
#########################################################

use strict;
# getopts package
use Getopt::Std;
use File::Copy;

#########################################################
# temporaries
#########################################################
my @TOREMOVE = ();

#########################################################
# Defaults
#########################################################

my $DFG2OTTER = "dfg2otter";    # dfg2otterpath
my $OUTFILESUFFIX = ".ot";      # generic output file suffix
my $ERRORFILESUFFIX = ".error"; # generic error file suffix

#########################################################
# options
#########################################################

# define default options
my %options = ( t => 10 );    # default: 10 seconds timeout for proof checker

# read options
&getopts("t:", \%options);

# shortcuts for options
my $timelimit = $options{'t'};

#########################################################
# MAIN
#########################################################
$SIG{INT} = \&catchint;         # set up interrupt handler
# try to get dfg2otter path from environment
# otherwise default is kept
if (defined($ENV{'DFG2OTTER'})) { 
  $DFG2OTTER = $ENV{'DFG2OTTER'};
}


# check that there are at most 2 arguments
if (@ARGV > 2) { die "Too many arguments.\n";}
#########################################################
# if there are no arguments, just call the original
#########################################################
if (@ARGV == 0) {
  system("$DFG2OTTER -Stdin");
}

#########################################################
# if one argument is given, use it as input file and
# generate file.ot output file 
#########################################################
my ($in, $out);
if (@ARGV == 1) {
  $in  = $ARGV[0];
  $out = $in."_".$$."_".$OUTFILESUFFIX;
  print `$DFG2OTTER -Stdin < $in > $out`;
}

#########################################################
# if two arguments are given, generate output filename
# first, write parameters to it, and then append dfg2otter
# output
#########################################################
my ($errorfile, $parout, $convout, @conv_res, $errorlines);

if (@ARGV == 2) { 
  
  $in  = $ARGV[0];
  $out = $ARGV[1];
  $errorfile = $out."_".$$."_".$ERRORFILESUFFIX;
  
  # print parameters
  $parout  = gettmp($out);
  open(CONV, ">$parout") or die "dfg2otter.pl: could not open '$in': $!"; 
  select CONV;
  &print_parameters();
  close CONV;
  
  # convert SPASS file
  $convout = gettmp($out);
  @conv_res = `$DFG2OTTER $in $convout 2>$errorfile`;
  
  # echo STDERR, remove files if errors occured
  # (that is if $errorfile is not empty)
  if (-e $errorfile) {
    unless (open(EF, "<$errorfile")) {
      cleanup();
      die "dfg2otter.pl: error opening '$errorfile': $!";
    }
    $errorlines = 0;
    while (<EF>) {
      print STDERR;
      $errorlines++;
    }
    close(EF);
    unlink($errorfile);
    if ($errorlines) { 
      unlink($out);
    }
  }
  
  # the following lines copy the file $parout to $out, and then append $convout,
  # like the UNIX command 'cat $parout $convout > $out'
  copy($parout, $out) or die "Could not copy $parout to $out: $!";
  open (IN, "$convout") or die "Could not open $convout: $!";
  open(APPEND, ">>$out") or die "Could not open $out: $!";
  while (<IN>) {
    print APPEND;
  }
  close(APPEND);
  close(IN);
  
  cleanup();
}


####################################################
# INPUT  : A signal name
# EFFECTS: Remove temporaries and reports interrupt 
####################################################
sub catchint {
  my $signame = shift;
  cleanup();
  die "\n\ndfg2otter.pl: caught signal $signame. Cleaning up."; 
}

####################################################
# EFFECTS: Remove temporaries
#####################################################
sub cleanup {
  unlink @TOREMOVE;
}

####################################################
# EFFECTS: Print otter default parameters to stdout
####################################################
sub print_parameters {
  print "set(process_input).\n";
  print "set(binary_res).\n";
  print "set(factor).\n";
  print "set(hyper_res).\n";
  print "assign(pick_given_ratio, 4).\n";
  print "clear(print_kept).\n";
  print "clear(print_new_demod).\n";
  print "clear(print_back_demod).\n";
  print "clear(print_back_sub).\n";
  print "set(knuth_bendix).\n";
  
  print "set(para_from).\n";
  print "set(para_into).\n";
  print "set(para_from_right).\n";
  print "set(para_into_right).\n";
  print "set(para_from_left).\n";
  print "set(para_into_left).\n";
  print "set(para_from_vars).\n";
  print "set(back_demod).\n";
  print "set(auto).\n";
  print "assign(max_seconds, $timelimit).\n";
}

####################################################
# INPUT:  A filename
# OUTPUT: A similar filename that is unique 
####################################################
sub gettmp {
  my ($file) = @_;
  my ($cnt)  = 0;
  
  while (-e $file."$cnt") {
    $cnt++;
  }
  $file = $file."$cnt"; 
  unshift(@TOREMOVE, $file); 
  
  return($file);
}
